test/e2e/config/aws-dev.yaml

```diff
@@ -0,0 +1,104 @@
+---
+# E2E test scenario using local dev images and manifests built from the source tree for following providers:
+# - cluster-api
+# - bootstrap kubeadm
+# - control-plane kubeadm
+# - aws
+
+# For creating local dev images built from the source tree;
+# - from the CAPI repository root, `make docker-build REGISTRY=gcr.io/k8s-staging-cluster-api` to build the cluster-api, bootstrap kubeadm, control-plane kubeadm provider images.
+# - from the CAPA repository root, `make docker-build REGISTRY=gcr.io/k8s-staging-cluster-api` to build the aws provider image.
+
+# Following env variables should be set before running the provider
+
+# export AWS_REGION=
+# export AWS_SSH_KEY_NAME=
+# export AWS_B64ENCODED_CREDENTIALS=
+
+images:
+# Use local dev images built source tree;
+- name: gcr.io/k8s-staging-cluster-api/cluster-api-controller-amd64:dev
+  loadBehavior: mustLoad
+- name: gcr.io/k8s-staging-cluster-api/kubeadm-bootstrap-controller-amd64:dev
+  loadBehavior: mustLoad
+- name: gcr.io/k8s-staging-cluster-api/kubeadm-control-plane-controller-amd64:dev
+  loadBehavior: mustLoad
+- name: gcr.io/k8s-staging-cluster-api/cluster-api-aws-controller-amd64:dev
```

Comment:
If this is attempting to test latest capi vs capa, should this target a known version of capa instead? With the current configuration it would be difficult to know whether it was a change in capi or capa that caused the test to break without digging in further.

Reply:
This is a dev config file designed to use what is in your local folder, so you control what to test/what is changed using git.

If instead, we would like to schedule the jobs in ci we should create a different config (similar to what I have done for docker-dev.yaml/docker-ci.yaml), and in that case, we can decide to pin or CAPI or CAPA versions. 

FYI CAPV is going to import these jobs specs -that are exposed as a library function - and manage a test config file and the prow scheduling from the CAPV repo.

Reply:
It is using locally built images that needs to be built following the comments at the beginning of this config.

Reply:
Didn't see @fabriziopandini's comment. You can ignore my comment above.